main.tf

```
@@ -234,6 +234,172 @@ module "cloudwatch_metrics" {
   tags = var.tags
 }
 
+################################################################################
+# EFS CSI DRIVER
+################################################################################
+
+locals {
+  efs_csi_driver_service_account = try(var.efs_csi_driver.service_account_name, "efs-csi-controller-sa")
+}
+
+data "aws_iam_policy_document" "aws_efs_csi_driver" {
+  statement {
+    sid       = "AllowDescribeAvailabilityZones"
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "ec2:DescribeAvailabilityZones",
+    ]
+  }
+
+  statement {
+    sid    = "AllowDescribeFileSystems"
+    effect = "Allow"
+    resources = [
+      "arn:${local.partition}:elasticfilesystem:${local.region}:${local.account_id}:file-system/*",
+      "arn:${local.partition}:elasticfilesystem:${local.region}:${local.account_id}:access-point/*"
+    ]
+
+    actions = [
+      "elasticfilesystem:DescribeAccessPoints",
+      "elasticfilesystem:DescribeFileSystems",
+      "elasticfilesystem:DescribeMountTargets"
+    ]
+  }
+
+  statement {
+    sid       = "AllowCreateAccessPoint"
+    effect    = "Allow"
+    resources = ["arn:${local.partition}:elasticfilesystem:${local.region}:${local.account_id}:file-system/*"]
+    actions   = ["elasticfilesystem:CreateAccessPoint"]
+
+    condition {
+      test     = "StringLike"
+      variable = "aws:RequestTag/efs.csi.aws.com/cluster"
+      values   = ["true"]
+    }
+  }
+
+  statement {
+    sid       = "AllowDeleteAccessPoint"
+    effect    = "Allow"
+    resources = ["arn:${local.partition}:elasticfilesystem:${local.region}:${local.account_id}:access-point/*"]
+    actions   = ["elasticfilesystem:DeleteAccessPoint"]
+
+    condition {
+      test     = "StringLike"
+      variable = "aws:ResourceTag/efs.csi.aws.com/cluster"
+      values   = ["true"]
+    }
+  }
+
+  statement {
+    actions = [
+      "elasticfilesystem:ClientRootAccess",
+      "elasticfilesystem:ClientWrite",
+      "elasticfilesystem:ClientMount",
+    ]
+    resources = ["arn:${local.partition}:elasticfilesystem:${local.region}:${local.account_id}:file-system/*"]
+    condition {
+      test     = "Bool"
+      variable = "elasticfilesystem:AccessedViaMountTarget"
+      values   = ["true"]
+    }
+  }
+}
+
+resource "aws_iam_policy" "aws_efs_csi_driver" {
+  count = var.enable_efs_csi_driver ? 1 : 0
+
+  name        = "${var.cluster_name}-efs-csi-driver"
+  description = "IAM Policy for AWS EFS CSI Driver"
+  policy      = data.aws_iam_policy_document.aws_efs_csi_driver.json
```

then flip this to use the index based on the conditional 
```suggestion
  policy      = data.aws_iam_policy_document.aws_efs_csi_driver[0].json
```